image/oci.go

```diff
@@ -187,7 +188,18 @@ func (m *manifestOCI1) convertToManifestSchema2() (types.Image, error) {
 	layers := make([]manifest.Schema2Descriptor, len(m.m.Layers))
 	for idx := range layers {
 		layers[idx] = schema2DescriptorFromOCI1Descriptor(m.m.Layers[idx])
-		layers[idx].MediaType = manifest.DockerV2Schema2LayerMediaType
+		switch layers[idx].MediaType {
+		case imgspecv1.MediaTypeImageLayerNonDistributable:
```

Comment:
Note that https://github.com/opencontainers/image-spec/pull/788 adds also the compressed variants for non-distributable layers. Are there (should there be) corresponding schema2 variants as well?

Reply:
I have mixed feelings about that. We're already lying about schema2 knowing about zstd, so it would make sense to add a foreign zstd type as well.

However, I am not sure if we shouldn't just throw zstd-errors for schema2 images and focus on OCI?

Reply:
I added them for now and updated the code conversion and update paths to take them into consideration as well.

Reply:
Arguably we should not be compressing schema2 images to ZSTD in the first place, at least until the official schema2 specification adds that.

OTOH that’s yet more infrastructure to build, and to an extent defeats the point of adding the feature, as long as OCI is not the prevalent format.

I don’t know; it’s just a thing that I don’t think has been actually discussed, and probably should have been.

Cc: @giuseppe

Reply:
Shall we error out for now? I don't want to add on-the-fly recompression to this PR as we need to get that in rather soon. 

(@giuseppe is on PTO this week.)